### PR TITLE
♻️🔨 Unexpected `mypy` upgrade revealed configuration and code failures

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

The issue was caused by our CI pipeline not using a pinned version of `mypy`; instead, it was always fetching the latest version. This happened because `requirements/ci.txt` did not include `requirements/tools.txt`, which specifies the version for `mypy`.

As a result, `mypy` was updated between the CI passing for this [PR](https://github.com/ITISFoundation/osparc-simcore/pull/6512) and its merge into the master branch (commit da15addd9126d4f2574ab16dd131bcf4d4c52206). 

The new version of `mypy` introduced additional type checks that uncovered issues in some parts of the code. Specifically, we had to make further changes to the section handling invoice attachments in emails to resolve these newly identified issues. 

### Highlights

- ♻️ `payments` service: 
    - Proper attachment and better error handling to invoice attachment. 
    - Adapts tests
    - @matusdrobuliak66  please pay special attention to the changes i did of your code in 
          -  `services/payments/src/simcore_service_payments/services/notifier_email.py`
          -  `services/payments/tests/unit/test_services_notifier_email.py`
- ♻️ ensures `create_troubleshotting_log_message` does not raise
- 🔨 uses `uv` run to run script

## Related issue/s

- Fixes [typecheck in CI pipelines](https://github.com/ITISFoundation/osparc-simcore/actions/runs/11327310701/job/31502541898#step:9:20) 

## Dev-ops checklist
NOne

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Review by GitHK — APPROVED** [review]

**Review by jsaq007 — COMMENTED** [review]
👍

**Review by matusdrobuliak66 — APPROVED** [review]

**Review by giancarloromeo — APPROVED** [review]

File: packages/service-library/src/servicelib/utils_secrets.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def _is_possibly_sensitive(name: str, sensitive_keywords: set[str]) -> bool:
 
 
 def mask_sensitive_data(
-    data: dict, *, extra_sensitive_keywords: set[str] | None = None
+    data: dict[str, Any], *, extra_sensitive_keywords: set[str] | None = None
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/utils_secrets.py:71]: Actually wanted to ask about the use of this function with regards to the e2e tests.
The point was actually to be able to copy/paste from the e2e logs directly the code to re-run locally. is it still possible with the use of this?

**Review by sanderegg — APPROVED** [review]

File: packages/service-library/src/servicelib/utils_secrets.py (hunk 1 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/utils_secrets.py:71]: @sanderegg lets check this offline. It risks of ending up in a "spaces vs tabs" discussion :-)

**Pull request merged**